userappwithbroker/src/main/java/com/microsoft/aad/adal/example/userappwithbroker/MainActivity.java

```diff
@@ -292,23 +296,28 @@ public void onError(Exception exc) {
      */
     private void saveUserIdFromAuthenticationResult(final AuthenticationResult authResult, final String authority) {
         mSharedPreference = getSharedPreferences(SHARED_PREFERENCE_STORE_USER_UNIQUEID, MODE_PRIVATE);
+        if (null != authResult) {
```

Comment:
I don't necessarily have a problem with defensively checking if this parameter is `null`, but I will call out that if `onSuccess()` was called with a `null` result, we wouldn't make it this far: it would throw an NPE inside of `onSuccess()` when we log out the accessToken. You could omit this check, since this code should never get hit.

Reply:
Thank you, Brian. Yes, I understand. I was thinking it would be better to check the input object for future usage because it is a single method (I understand it is private method and currently just get called in the onSuccess which has null check on result) and has getAuthority and getUserInfo called on the object. If the other calling app does not has the null check on the input, nullpointerexception will be thrown without the checking.